runtime/oti/vm_api.h

```diff
@@ -4669,6 +4690,41 @@ void
 releaseVThreadInspector(J9VMThread *currentThread, jobject thread);
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
+#if JAVA_SPEC_VERSION >= 24
+/**
+ * @brief Attach all yielded VirtualThread monitors to carrier thread.
+ *
+ * @param currentThread the current thread
+ * @param continuationObject the Continuation object
+ */
+void
+preparePinnedVirtualThreadForMount(J9VMThread *currentThread, j9object_t continuationObject);
+
+/**
+ * @brief Inflate all monitors and prepare the VirtualThread to yield.
+ *
+ * @param currentThread the current thread
+ * @param syncObj object to block/wait on
+ * @param isObjectWait if the call is from Object.wait()
+ *
+ * @return J9_OBJECT_MONITOR_YIELD_VIRTUAL if the can be successfully yielded;
+ * otherwise, an error code is returned
+ */
+UDATA
+preparePinnedVirtualThreadForUnmount(J9VMThread *currentThread, j9object_t syncObj, BOOLEAN isObjectWait);
+
+/**
+ * @brief Find a list of virtual thread to be unblocked. This is a helper method for
+ * JVM_TakeVirtualThreadListToUnblock (see javanextvmi.cpp).
+ *
+ * @param currentThread the current thread
+ * @param vm pointer to J9JavaVM
+ *
+ * @return a list of virtual threads to be unblocked
+ */
+jobject
+takeVirtualThreadListToUnblock(J9VMThread *currentThread, J9JavaVM *vm);
```

Comment:
Isn't the `vm` parameter redundant here (it can be derived from `currentThread`)?

Reply:
The callee already has both parameters. Passing J9JavaVM as an input parameter avoids the need to dereference it twice from currentThread.

Reply:
I assume you meant "caller" (not "callee"), but that doesn't necessarily mean it's better. Accessing a separate parameter instead of dereferencing the first parameter may or may not have better performance. I'd classify it (at best) premature optimization.